src/virtual_dom/vlist.rs

```diff
@@ -103,22 +108,52 @@ impl VDiff for VList {
         }
 
         // Process children
-        let mut lefts = self.children.iter_mut();
-        let mut rights = rights.drain(..);
-        loop {
-            match (lefts.next(), rights.next()) {
-                (Some(left), Some(right)) => {
-                    previous_sibling = left.apply(parent, previous_sibling.as_ref(), Some(right));
+        let lefts = self.children.iter_mut();
+        let key_count = rights.iter().filter(|r| r.key().is_some()).count();
+        let mut rights_nokeys = Vec::with_capacity(rights.len() - key_count);
+        let mut rights_lookup = HashMap::with_capacity(key_count);
+        for r in rights.drain(..) {
+            if let Some(key) = r.key() {
+                if rights_lookup.contains_key(&key) {
+                    log::error!("Duplicate key of {}", &key);
```

Comment:
I think we should check for this for `lefts` instead of `rights`. Otherwise the error will be delayed until the next render.

Reply:
I added a check to put the keys in a hashset and check and see if there are duplicates.   Shows an error if there is one.  I couldn't think of a way to switch going just through the left keys since they need to be in order.

Reply:
This is exactly what I expected, looks good!